Documentation/marconi.rst

```diff
@@ -0,0 +1,277 @@
+  --disable-dap
+
+  $ make -j 
+  $ make install
+  $ cd ..
+
+8. Installing ParMetis:
+
+::
+
+  $ wget http://glaros.dtc.umn.edu/gkhome/fetch/sw/parmetis/parmetis-4.0.3.tar.gz
+  $ tar -xvf ./parmetis-4.0.3.tar.gz
+  $ cd parmetis-4.0.3
+  #edit ./metis/include/metis.h IDXTYPEWIDTH to be 64 (default is 32).
+  $ make config cc=mpicc cxx=mpiCC prefix=$SEISSOL_INSTALL
+  $ make install
+  $ cp build/Linux-ppc64le/libmetis/libmetis.a $SEISSOL_INSTALL/lib
+  $ cp metis/include/metis.h $SEISSOL_INSTALL/include
+  $ cd ..
+
+9. Install GemmForge. Please, follow steps described :ref:`here <gemmforge_installation>`. 
+
+10. Install SeisSol:
+
+::
+
+  $ module load cuda/11.0
+  $ git clone --recurse-submodules https://github.com/SeisSol/SeisSol.git
+  $ cd SeisSol
+  $ mkdir build && cd build
+
+  $ CC=mpicc CXX=mpicxx FC=mpifort cmake .. \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DDEVICE_ARCH=nvidia \
+  -DDEVICE_SUB_ARCH=sm_70 \
+  -DHOST_ARCH=power9 \
+  -DPRECISION=single \
+  -DCOMMTHREAD=ON
+
+  $ make -j
+
+11. Run SeisSol-proxy as a sanity check:
+
+::
+
+  ./SeisSol_proxy_Release_snvidia_6_elastic 100000 100 all
+
+
+Running SeisSol
+---------------
+
+As discussed :ref:`here <gpu_process_pinning>`, process pinning is important for SeisSol GPU version.
+IBM Power9 is an example of RISC architecture designed with with 4-way hyperthreading and 8 cores per CPU.
+In total, each node of Marconi 100 can run 256 threads. By and large process pinning needs a special 
+care on such architectures because some libraries have different meanings of cores and threads.
+
+Below you can see an example of a *batch script* with parameters resulting in an optimal process pinning.
+Note that each node of Marconi 100 has 2 Mellanox network cards i.e., each per NUMA domain. In this example,
+we enforce UCX to utilize both. Moreover, we reserve one 1 core for each MPI process for SeisSol communication thread.
+
+In this particular case it is not necessary to provide a number of processes after **mpirun** because OpenMPI was compiled 
+with PMIX (see step 5).
+ 
+
+::
+
+  $ cat ./job.sh
+  #!/bin/bash
+  #SBATCH --account=<you account>
+  #SBATCH --partition=m100_usr_prod
+  #SBATCH --qos=m100_qos_dbg
+  #SBATCH --time <time>
+  #SBATCH --nodes <number of nodes>
+  #SBATCH --ntasks-per-node=4
+  #SBATCH --cpus-per-task=32
+  #SBATCH --gres=gpu:4
+  #STABCH --gpu-bind=closest
+  #SBATCH --mem=161070
+  #SBATCH --job-name=<your job name>
+  #SBATCH --mail-type=ALL
+  #SBATCH --mail-user=<user_email>
+  #SBATCH --output=seissol.out
+  #SBATCH --error=seissol.err
+  #SBATCH --exclusive
+
+  NUM_CORES=$(expr $SLURM_CPUS_PER_TASK / 4)
+  NUM_COMPUTE_CORES=$(expr $NUM_CORES - 1)
+
+  export OMP_NUM_THREADS=$NUM_COMPUTE_CORES
+  export OMP_PLACES="cores($NUM_COMPUTE_CORES)"
+  export PROC_BIND=spread
+
+  export DEVICE_STACK_MEM_SIZE=1.5
+  export UCX_MEMTYPE_CACHE=n
+  
+  mpirun --report-bindings --map-by ppr:$SLURM_NTASKS_PER_NODE:node:pe=$NUM_CORES \
```

Comment:
isn't -np missing?

Reply:
Hi, 

Thanks for noticing. No it is not missing __in this particular case__ because OpenMPI on M100 was compiled with PMIX3.

```
--with-pmix=/opt/pmix/3.1.5
```

 As far as I understand PMIx is an API (or a protocol) which sits between a resource manager (e.g., slurm, torque) and an MPI implementation. In short, a resource manager sends all necessary info to PMIx and then it is delivered to MPI. This information can include hostnames of each involved compute node, maybe a mapping between PIDs and ranks, etc. 

I know that it is super specific and thus there is a line, right above the batch file, saying

```
In this particular case it is not necessary to provide a number of processes after **mpirun** because OpenMPI was compiled with PMIX (see step 5).
```